packages/client/src/client.js

```diff
@@ -526,6 +529,12 @@ export class PercyClient {
         return 'web';
     }
   }
+
+  computeSkipDiscoveryState() {
+    const type = this.tokenType();
```

Comment:
lets pass an argument to `tokenType` which returns token without raising error, by default it will work how it is working currently.

Reply:
also lets move this function to `core` since `skipDiscovery` is part of core

and rename it `shouldSkipAssetDiscovery()`